### PR TITLE
HTML report: make sysinfo section aware of pre/post/profile

### PR DESCRIPTION
**Pull request by clebergnu**
Currently the sysinfo info on the HTML report only shows the pre-testp hase of sysinfo.  It should really include all three phases.
    
Reference: https://trello.com/c/sXZQTfta

### CODE REVIEW

File: avocado/core/runner.py [hunk 1]
```diff
@@ -466,9 +466,9 @@ def run_suite(self, test_suite, mux, timeout=0, replay_map=None,
             TEST_LOG.error('Job interrupted by ctrl+c.')
             summary.add('INTERRUPTED')
 
-        self.result.end_tests()
-        self.job.funcatexit.run()
         if self.job.sysinfo is not None:
             self.job.sysinfo.end_job_hook()
+        self.result.end_tests()
+        self.job.funcatexit.run()
```
Comment (ldoktor) [inline avocado/core/runner.py]: Actually the `end_tests` method means all tests finished, but based on the usage, it's more like `end_job` (it's the last interface between results and job). So if I think about this more, probably even the `job.funcatexit` should be executed before we produce the `end_tests` (which in the ideal world would be `end_job`, but it's too late to change it now).

File: avocado/core/html.py [hunk 2]
```diff
@@ -115,7 +115,7 @@ def _get_sysinfo(self, sysinfo_file):
         return sysinfo_contents
 
     def hostname(self):
-        return self._get_sysinfo('hostname')
+        return self._get_sysinfo('hostname').strip()
```
Comment (ldoktor) [inline avocado/core/html.py]: good catch

Comment (ldoktor) [inline avocado/core/html.py]: Btw how about putting the `stript` directly to the `_get_sysinfo`. Aren't all of the sysinfo alike?

Comment (clebergnu) [inline avocado/core/html.py]: The thing is that, maybe, the trailing newline is indeed part of the *file* or *command* that was collected.  The hostname is extracted from a file, but we're not presenting it as the contents of a *file* or *command*... does it make sense?

File: avocado/core/resources/htmlresult/templates/report.mustache [hunk 3]
```diff
@@ -77,31 +77,101 @@
...
                                 </div>
                             </div>
                         </div>
                     </div>
+                    <!-- Profilers end -->
+
                 </div>
+            </div>
+            <!-- Sysinfo end -->
+
```
Comment (ldoktor) [inline avocado/core/resources/htmlresult/templates/report.mustache]: Sounds good, let me check how it behaves...

File: avocado/core/html.py (hunk 2 above)
Comment (ldoktor) [inline avocado/core/html.py]: It does, that's why I was asking and not demanding :smile:. OK so let's be sure and use it only for the hostname for now.

**Comment by ldoktor** [comment]
Yes, it's more consistent and it behaves nicely. @clebergnu please let me know what you think regarding the `results.end_tests`, I'm prepared to merge this, when we sort out this little detail.

**Comment by clebergnu** [comment]
Yep, I know you were not demanding it... And I was honestly asking so you could help me to validate the logic I was proposing. Now, the end_tests thing... I'll leave that to the morning. Thanks for the review so far!

**Comment by ldoktor** [comment]
Sure, we are not in a hurry.

PS: It is morning here, so are you ready :stuck_out_tongue_winking_eye:

File: avocado/core/runner.py (hunk 1 above)
Comment (clebergnu) [inline avocado/core/runner.py]: So, I was thinking about this... It's a tough call.  It really depends on how he define and document the `funcatexit` feature.  Right now, this is what we say about them:

"It’s possible to register a callback function that will be called when all the tests have finished running"

So it's accurate in that sense.  All tests have finished running... and IMHO, having the results generated are part of that in a greater sense.

So, right now, I'd leave it as is.

Comment (ldoktor) [inline avocado/core/runner.py]: OK

**Pull request merged**